### ISSUE TITLE
Profiling Models with Unity Profiler

### ISSUE THREAD

**Issue by roger-creus** [Unity-Technologies/barracuda-release#209]
Hello :)

I have run a mobile application that integrates a ONNX model in an Android device and have successfully profiled it in a PC with the Unity Profiler tool. However, I would like to see the specific time (or memory cost) spent in the model only. I have seen works that are even capable of retrieving the spent resources in each layer of the model. How can this be achieved?

Thank you!

**Comment by amirebrahimi** [comment]
I think there are a few things you could try. Have you tried [remote profiling of your Android build](https://docs.unity3d.com/Manual/profiler-profiling-applications.html)?

You can also [add your own profile markers](https://docs.unity3d.com/ScriptReference/Profiling.Profiler.BeginSample.html) around your model execution code.

Another option, although this won't help on mobile (i.e. Editor only) is to try using [`ModelExecutionsReporter`](https://docs.unity3d.com/Packages/com.unity.barracuda@2.1/api/Unity.Barracuda.ModelExecutionsReporter.html) by passing it to `CreateWorker` as an argument. You need to set `ENABLE_BARRACUDA_STATS` in your project defines for this to work.

**Comment by roger-creus** [comment]
> I think there are a few things you could try. Have you tried [remote profiling of your Android build](https://docs.unity3d.com/Manual/profiler-profiling-applications.html)?
> 
> You can also [add your own profile markers](https://docs.unity3d.com/ScriptReference/Profiling.Profiler.BeginSample.html) around your model execution code.
> 
> Another option, although this won't help on mobile (i.e. Editor only) is to try using [`ModelExecutionsReporter`](https://docs.unity3d.com/Packages/com.unity.barracuda@2.1/api/Unity.Barracuda.ModelExecutionsReporter.html) by passing it to `CreateWorker` as an argument. You need to set `ENABLE_BARRACUDA_STATS` in your project defines for this to work.

Running remote profiling of the Android Build is exactly what I have run. It works well! my question relates how can I see the model-specific profile in the Profiler Window (i.e. resources spent in each layer of a CNN model).

**Comment by AlexRibard** [comment]
Have you tried profiling with [Renderdoc](https://renderdoc.org/docs/how/how_android_capture.html)? It should give you detailed information about every layer and how long it takes/associated resources

**Comment by amirebrahimi** [comment]
@yuyecreus - In addition to Alex's suggestion did you see the "Barracuda.\<opname\>" markers in the profiler? If you're only running one model then it should be easy to see how much time each op is taking.

**Comment by roger-creus** [comment]
> @yuyecreus - In addition to Alex's suggestion did you see the "Barracuda.<opname>" markers in the profiler? If you're only running one model then it should be easy to see how much time each op is taking.

Sure now I found it. The thing is the "Barracuda.opname" markers only appear in the hierarchy at the moment in time that Barracuda is used. As I just use a model when the user clicks a button, I had to find the frame in which the model was used (easily found because there is a huge maxima of resources spent there). All layers can be analyzed separately, which is very cool.
